src/App/ConnectionPage.xaml.cs

```diff
@@ -157,29 +175,61 @@ private async void ShowConnectFailure(string ipStr)
 
             _ = await failedConnectDialog.ShowAsync();
         }
-
-        private void IpTextBox_TextChanged(object sender, TextChangedEventArgs e)
-        {
-            if(!string.IsNullOrWhiteSpace(IpTextBox.Text))
-            {
-                ConnectButton.IsEnabled = true;
-            }
-        }
-
-        private void IpTextBox_KeyDown(object sender, KeyRoutedEventArgs e)
-        {
-            if (e.Key == Windows.System.VirtualKey.Enter)
-            {
-                if (!String.IsNullOrWhiteSpace(IpTextBox.Text))
-                {
-                    ConnectButton_Click(null, null);
-                }
-            }
-        }
-
+        
         private void AboutButton_Click(object sender, RoutedEventArgs e)
         {
             this.Frame.Navigate(typeof(AboutPage));
+        }
+
+        List<TextBox> GetAllTextBoxes(DependencyObject parent)
+        {
+            var list = new List<TextBox>();
+            for (int i = 0; i < VisualTreeHelper.GetChildrenCount(parent); i++)
+            {
+                var child = VisualTreeHelper.GetChild(parent, i);
+                if (child is TextBox)
+                    list.Add(child as TextBox);
+                list.AddRange(GetAllTextBoxes(child));
+            }
+            return list;
+        }
+
+        private void TextBox_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            var allTextBoxes = GetAllTextBoxes(this);
+            bool validData = true;
+            foreach (TextBox textbox in allTextBoxes)
+            {
+                if (String.IsNullOrWhiteSpace(textbox.Text))
+                {
+                    validData = false;
+                }
+            }
+            if (validData)
+            {
+                ConnectButton.IsEnabled = true;
+            }
+
+        }
+        
+        private void TextBox_KeyDown(object sender, KeyRoutedEventArgs e)

```

Comment:
I assume TextBox_TextChanged fires before TextBox_KeyDown (in any valid input scenario).

Therefore, could you do this?

```
if (ConnectButton.IsEnabled && e.Key == Windows.System.VirtualKey.Enter)
            {
                ConnectButton_Click(null, null);
            }

```